contracts/src/v0.8/test/HeartbeatRequester.t.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "forge-std/Test.sol";
+import {HeartbeatRequester} from "../HeartbeatRequester.sol";
+import "../mocks/MockAggregator.sol";
+import "../mocks/MockAggregatorProxy.sol";
+import "../mocks/MockOffchainAggregator.sol";
+
+// from contracts directory,
+// forge test --match-path src/v0.8/test/HeartbeatRequester.t.sol
+
+contract HeartbeatRequesterTest is Test {
+  HeartbeatRequester public heartbeatRequester;
+  MockAggregator public aggregator;
+  IAggregatorProxy public aggregatorProxy;
+  MockAggregator public aggregator2;
+  IAggregatorProxy public aggregatorProxy2;
+
+  address internal constant USER = address(2);
+
+  event HeartbeatPermitted(address indexed permittedCaller, address newProxy, address oldProxy);
+  event HeartbeatRemoved(address indexed permittedCaller, address removedProxy);
+  error HeartbeatNotPermitted();
+
+  function setUp() public {
+    heartbeatRequester = new HeartbeatRequester();
+    aggregator = new MockAggregator();
+    aggregatorProxy = IAggregatorProxy(new MockAggregatorProxy(address(aggregator)));
+    aggregator2 = new MockAggregator();
+    aggregatorProxy2 = IAggregatorProxy(new MockAggregatorProxy(address(aggregator2)));
+  }
+
+  function test_permitHeartbeat_ArbitraryUser() public {
```

Comment:
If this is a test that should end up in the shared gas snapshot, you could add `_gas` at the end. Not saying this should be done here. It should only be done for tests where you really case about tracking its gas usage. All tests that use _gas should also implement `vm.pauseGasMetering();` and `vm.resumeGasMetering();` to improve accuracy. Again, for examples see CCIP